packages/@okta/vuepress-theme-prose/components/TermsAndConditionsDialog.vue

```diff
@@ -0,0 +1,267 @@
+        <div class="gdpr-condition">
+          <input
+            type="checkbox"
+            name=""
+            id="okta-contact"
+            ref="gdprBoxEl"
+            checked="false"
+          />
+          <label for="okta-contact">
+            I agree that Okta may contact me with marketing communications. See
+            Privacy Policy for details on how to unsubscribe.
+          </label>
+        </div>
+      </div>
+    </div>
+    <template v-slot:footer>
+      <div class="agree-policy" v-show="selectedCountry">
+        <div class="splitter"></div>
+        <p>
+          By clicking "continue", I agree to the applicable Free Trial terms in
+          <SmartLink :item="{ link: '/terms/' }">Okta's Terms</SmartLink> and
+          the processing of my personal data by Okta as described in the
+          <SmartLink :item="{ link: '/privacy-policy/' }"
+            >Privacy Policy</SmartLink
+          >.
+        </p>
+      </div>
+      <div class="terms-conditions-btns">
+        <div class="btn-container">
+          <input
+            type="button"
+            class="btn red-button"
+            @click="!isDisabledSocialAuth && setTaCUrlAndRedirect()"
+            :class="{ 'btn-disabled': isDisabledSocialAuth }"
+            value="continue"
+            :disabled="isDisabledSocialAuth"
+          />
+        </div>
+        <div class="btn-container mr-15">
+          <input
+            type="button"
+            class="btn social-btn"
+            value="cancel"
+            @click="closeTermsConditionsDialog()"
+          />
+        </div>
+      </div>
+    </template>
+  </DialogWindow>
+</template>
+
+<script>
+import {
+  countriesList,
+  americanStates,
+  canadaProvinces,
+  GDPR_COUNTRIES
+} from "../const/signup.const";
+
+const CANADA = "Canada";
+const USA = "United States";
+
+export default {
+  name: "TermsAndConditionsDialog",
+  components: {
+    SmartLink: () => import("./SmartLink"),
+    DialogWindow: () => import("./DialogWindow")
+  },
+  props: {
+    socialUrl: {
+      required: true,
+      type: String
+    }
+  },
+  data() {
+    return {
+      selectedCountry: "",
+      selectedRegion: "",
+      regionData: { type: "", list: [] }
+    };
+  },
+  computed: {
+    isDisabledSocialAuth() {
+      // If country not selected, disable "continue" button.
+      if (!this.selectedCountry) {
+        return true;
+      }
+      // If country Canada or USA but region not selected, disable "continue" button.
+      if (this.selectedCountry === USA || this.selectedCountry === CANADA) {
+        if (!this.selectedRegion) {
+          return true;
+        }
+      }
+    },
+    region: {
+      get() {
+        return this.regionData;
+      },
+      set(country) {
+        if (country === USA) {
+          this.regionData.list = americanStates;
+          this.regionData.type = "State";
+        } else if (country === CANADA) {
+          this.regionData.list = canadaProvinces;
```

Comment:
nit: we should have consistent naming, either `canadianProvinces` or change above to `americaStates` (I prefer the first one)

Reply:
hm, Canada use province like the word the describe some territory and USA use state that have the same meaning

Reply:
Oh, I'm referring the first half of the variable, not `states` vs `provinces`

Reply:
Shaw is referring to `canada/america` vs `canadian/american`, you currently have them mixed up.

Reply:
my bad sorry =)